src/encoder/plugins/FdkaacEncoderPlugin.cxx

```
@@ -0,0 +1,263 @@
+#include "AudioFormat.hxx"
+#include "config/ConfigError.hxx"
+#include "util/NumberParser.hxx"
+#include "util/ReusableArray.hxx"
+#include "util/RuntimeError.hxx"
+
+#include <fdk-aac/aacenc_lib.h>
+
+#include <stdexcept>
+
+#include <assert.h>
+#include <string.h>
+
+class FdkaacEncoder final : public Encoder {
+	const AudioFormat audio_format;
+	HANDLE_AACENCODER *AacEncoder;
+	AACENC_InfoStruct *info;
+	int input_buffer_pos = 0;
+
+	ReusableArray<unsigned char, 32768> output_buffer;
+	ReusableArray<unsigned char, 32768> input_buffer;
+	unsigned char *output_begin = nullptr, *output_end = nullptr, *input_begin = nullptr;
+
+public:
+	FdkaacEncoder(const AudioFormat _audio_format,
+		    HANDLE_AACENCODER *_AacEncoder, AACENC_InfoStruct *_info)
+		:Encoder(false),audio_format(_audio_format),AacEncoder(_AacEncoder),info(_info) {}
+
+	~FdkaacEncoder() override;
+
+	/* virtual methods from class Encoder */
+	void Write(const void *data, size_t length) override;
+	size_t Read(void *dest, size_t length) override;
+};
+
+class PreparedFdkaacEncoder final : public PreparedEncoder {
+	HANDLE_AACENCODER *AacEncoder;
+	AUDIO_OBJECT_TYPE aot;
+	int bitrate;
+	int quality;
+	bool aacenc_afterburner;
+
+public:
+	PreparedFdkaacEncoder(const ConfigBlock &block);
+
+	/* virtual methods from class PreparedEncoder */
+	Encoder *Open(AudioFormat &audio_format) override;
+
+	const char *GetMimeType() const override {
+		fprintf(stderr, "GetMimeType\n");
+		return "audio/aac";
+	}
+};
+
+PreparedFdkaacEncoder::PreparedFdkaacEncoder(const ConfigBlock &block)
+{
+	const char *value;
+
+	value = block.GetBlockValue("aot", "lc");
+	if (value != nullptr) {
+		if(!strcmp(value, "lc")) {
+			aot=AOT_AAC_LC;
+		}else if(!strcmp(value, "he")) {
+			aot=AOT_SBR;
+		}else if(!strcmp(value, "hev2")) {
+			aot=AOT_PS;
+		}else if(!strcmp(value, "ld")) {
+			aot=AOT_ER_AAC_LD;
+		}else if(!strcmp(value, "eld")) {
+			aot=AOT_ER_AAC_ELD;
+		}else {
+			aot=AOT_AAC_LC;
```

If an invalid value is configured, this should not be silently ignored.